OneCommunityGlobal/HighestGoodNetworkApp
# Navneeth create see project management tab custom permission

# Description
![bug description](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/6d15e2ff-d01e-45ed-9916-f74a0ebf2905)


## Related PRs:
This PR is linked to the BE PR [#545](https://github.com/OneCommunityGlobal/HGNRest/pull/545).


## Main changes explained:
- Employing consistent naming of permissions in `src/utils/routePermissions.js`
- Deciding that when a volunteer account has both `See Project Management Tab (Full Functionality)` AND `See Project Management Tab (Add team members, upload new WBSs only)`, then the `See Project Management Tab (Full Functionality)` permission will have priority. I.e., the volunteer account will be able delete, edit, and create users. 
- Rendering the UI according to the permissions wherein an account with just the `See Project Management Tab (Add team members, upload new WBSs only)` permission will not be able to update, delete, or create new projects.

## How to test:
1. Follow the steps in the backend (BE PR [#545](https://github.com/OneCommunityGlobal/HGNRest/pull/545)) to run the relevant PR.
2. In the frontend, check into the current branch.
3. do `npm install` and `npm run start:local` to run this PR locally
4. Log in as Owner -> Other Links -> Permissions Management -> Manage User Permissions
5. Enter the username of an account that does not have the “See Project Management Tab (Full Functionality)” permission by default (i.e., a Volunteer account). 
6. Next to “See Project Management Tab (Full Functionality)” or “See Project Management Tab (Add team members, upload new WBSs only)”, click on Add. Then scroll down and click on Submit.
7. Log in as the volunteer account (if already logged in, make sure to log out and log back in). The “Other Links" button should appear with ONLY the “Projects” option visible/accessible.
8. The volunteer account can create, update, and delete projects, add WBSs, and add team members in the Project Management page (Projects) if "See Project Management Tab (Full Functionality" permission was added. If only the "See Project Management Tab (Add team members, upload new WBSs only)" permission was added, then the volunteer can do all of the above except for creating, updating, and deleting projects.

## Video on how to test (with steps):

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/687dd9dd-260b-4140-b345-25c186a84753
